client/app/reducers/ManageAvailabilityReducer.js

```diff
@@ -1,5 +1,67 @@
-const initialState = {};
+import { Map, List } from 'immutable';
+import moment from 'moment';
+import { isSameDay } from 'react-dates';
+import * as actionTypes from '../constants/ManageAvailabilityConstants';
 
-export default function manageAvailabilityReducer(state = initialState) {
-  return state;
+const initialState = new Map({
+  isOpen: true,
+  visibleMonth: moment().startOf('month'),
+  reservedDays: new List(),
+  blockedDays: new List(),
+  daysToAllow: new List(),
+  daysToBlock: new List(),
+});
+
+export const EDIT_VIEW_OPEN_HASH = 'manage-availability';
+
+const hasChanges = (state) =>
+      state.get('daysToAllow').size > 0 ||
+      state.get('daysToBlock').size > 0;
+
+const sameDayAs = (day1) => (day2) => isSameDay(day1, day2);
+
+const withDay = (days, day) => days.push(day);
+
+const withoutDay = (days, day) => days.filterNot(sameDayAs(day));
+
+const allowDay = (state, day) =>
```

Comment:
One idea would be to have one list for all block and allow actions, and then use a `reduce` function (`reduce` as in functional programming `reduce`, not necessarily as Redux `reducer`).

Example (in pseudo-code that resembles JavaScript but probably doesn't work):

```
events = [];
events.push({action: "block", value: 1}) // => [{action: "block", value: 1}]
events.push({action: "block", value: 2}) // => [{action: "block", value: 1}, {action: "block", value: 2}]
events.push({action: "allow", value: 1}) // => [{action: "block", value: 1}, {action: "block", value: 2}, {action: "allow", value: 1}]
```

Now, to calculate the changes, you could do something like:

```
changes = events.reduce(function(all_changes, event) {
  // Implementation omitted
  //
  // One wacky idea how to implement this: 
  // Map "blocks" to `-1`, "allows" to "1", then `sum` them and filter all zeros :)
}, [])

console.log(changes) // => [{action: "block", value: 2}]
```

To calculate the current view:

```
newReservedDays = blockedDays + changes
```